base/bucket_gocb.go

```diff
@@ -1337,28 +1377,188 @@ func (bucket CouchbaseBucketGoCB) IsKeyNotFoundError(err error) bool {
 }
 
 func (bucket CouchbaseBucketGoCB) DeleteDDoc(docname string) error {
-	LogPanic("Unimplemented method: DeleteDDoc()")
-	return nil
+
+	manager, err := bucket.getBucketManager()
+	if err != nil {
+		return err
+	}
+
+	return manager.RemoveDesignDocument(docname)
+
 }
 
 func (bucket CouchbaseBucketGoCB) View(ddoc, name string, params map[string]interface{}) (sgbucket.ViewResult, error) {
-	LogPanic("Unimplemented method: View()")
-	return sgbucket.ViewResult{}, nil
+
+	// Block until there is an available concurrent view op, release on function exit
+	bucket.waitForAvailViewOp()
+	defer bucket.releaseViewOp()
+
+	viewResult := sgbucket.ViewResult{}
+	viewResult.Rows = sgbucket.ViewRows{}
+
+	viewQuery := gocb.NewViewQuery(ddoc, name)
+
+	// convert params map to these params
+	applyViewQueryOptions(viewQuery, params)
+
+	goCbViewResult, err := bucket.ExecuteViewQuery(viewQuery)
+	if err != nil {
+		return viewResult, err
+	}
+
+	if goCbViewResult != nil {
+
+		viewResultMetrics, gotTotalRows := goCbViewResult.(gocb.ViewResultMetrics)
+		if gotTotalRows {
+			viewResult.TotalRows = viewResultMetrics.TotalRows()
+		} else {
+			Warn("Unable to type assert goCbViewResult -> gocb.ViewResultMetrics.  The total rows count may be inaccurate.")
+		}
+
+		for {
+
+			viewRow := sgbucket.ViewRow{}
+			if gotRow := goCbViewResult.Next(&viewRow); gotRow == false {
+				break
+			}
+
+			viewResult.Rows = append(viewResult.Rows, &viewRow)
+
+			// If goCbViewResult could not be type asserted to gocb.ViewResultMetrics, fallback to counting rows
+			if !gotTotalRows {
+				viewResult.TotalRows += 1
+			}
+
+		}
+
+		// Any error processing view results is returned on Close
+		err := goCbViewResult.Close()
+		if err != nil {
+			return viewResult, err
+		}
+	}
+
+	return viewResult, nil
+
 }
 
+
 func (bucket CouchbaseBucketGoCB) ViewCustom(ddoc, name string, params map[string]interface{}, vres interface{}) error {
-	LogPanic("Unimplemented method: ViewCustom()")
+
```

Comment:
Yeah, I was actually struggling with that and consciously decided to not attempt to factor out a common core method, which I decided would end up too difficult to understand.  Instead I tried to refactor out all common code (like `applyViewQueryOptions()`) into other methods to reduce duplication.  To that end, in 62fbb26e3604d5189dc300b7e225136704fcc27a I just factored out `getTotalRows()` into it's own method.  Now there is barely any code left in `View()` or `ViewCustom()`, they only contain calls out to other code!